src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/ApiDescriptionExtensions.cs

```diff
@@ -51,28 +54,55 @@ public static IEnumerable<object> CustomAttributes(this ApiDescription apiDescri
         internal static string RelativePathSansParameterConstraints(this ApiDescription apiDescription)
         {
             var routeTemplate = apiDescription.RelativePath;
-            
-            // We want to filter out qualifiers that indicate a constract (":")
+
+            // We want to filter out qualifiers that indicate a constraint (":")
             // a default value ("=") or an optional parameter ("?")
-            while (routeTemplate.IndexOfAny(new[] { ':', '=', '?' }) != -1)
+            if (routeTemplate.IndexOfAny(Disallowed) == -1)
+            {
+                // Quick-exit if there's nothing to remove.
+                return routeTemplate;
+            }
+
+            var modifiedTemplate = new StringBuilder(routeTemplate.Length);
+            var index = 0;
+            var isBetweenCurlyBraces = false;
+            while (index < routeTemplate.Length)
             {
-                var startIndex = routeTemplate.IndexOfAny(new[] { ':', '=', '?' }) ;
-                var tokenStart = startIndex + 1;
-                // Only find a single instance of a '}' after our start
-                // character to avoid capturing escaped curly braces
-                // in a regular expression constraint
+                var current = routeTemplate[index];
+
+                if (current == '{')
+                {
+                    isBetweenCurlyBraces = true;
+                    goto next;
+                }
+
+                if (current == '}')
+                {
+                    isBetweenCurlyBraces = false;
+                    goto next;
+                }
+
+                if (isBetweenCurlyBraces && Disallowed.Contains(current))
+                {
+                    // Only find a single instance of a '}' after our start
+                    // character to avoid capturing escaped curly braces
+                    // in a regular expression constraint
                 findEndBrace:
-                    var endIndex = routeTemplate.IndexOf('}', tokenStart);
-                    if (endIndex < routeTemplate.Length - 1 && routeTemplate[endIndex + 1] == '}')
+                    index = routeTemplate.IndexOf('}', index + 1);
+                    if (index < routeTemplate.Length - 1 && routeTemplate[index + 1] == '}')
                     {
-                        tokenStart = endIndex + 2;
+                        index += 2;
                         goto findEndBrace;
                     }
+                    continue;
+                }
 
-                routeTemplate = routeTemplate.Remove(startIndex, endIndex - startIndex);
+            next:
+                modifiedTemplate = modifiedTemplate.Append(current);
+                index += 1;
             }
-            
-            return routeTemplate;
+
+            return modifiedTemplate.ToString();
         }
     }
 }
```

Comment:
I feel this code is becoming increasingly complicated and brittle and wonder if we could simplify by leveraging the same route template parser that's used by ASP.NET Core routing (see https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.routing.template.templateparser?view=aspnetcore-6.0)? For example, something along the following lines:

```csharp
internal static string RelativePathSansParameterConstraints(this ApiDescription apiDescription)
{
    var routeTemplate = TemplateParser.Parse(apiDescription.RelativePath);

    var builder = new StringBuilder();
    foreach (var segment in routeTemplate.Segments)
    {
        foreach (var part in segment.Parts)
        {
            builder.Append(part.Name != null ? $"{{{part.Name}}}" : part.Text);
        }
        builder.Append("/");
    }
    
    return builder.ToString().TrimEnd('/');
}
```

Reply:
Sure thing. (I admit that I was minimally solving my immediate problem.) Should I work something like that sample into this MR, or would you prefer to handle it yourself?

Reply:
If you could work it into this PR that would be great 👍